src/config/scrivitoSites.ts

```diff
@@ -84,24 +84,33 @@ export function isNoSitePresent(): boolean {
 }
 
 export async function ensureSiteIsPresent() {
-  if ((await load(currentSiteId)) === null) {
-    if (await load(isNoSitePresent)) {
-      ensureUserIsLoggedIn()
-      return
-    }
-
-    navigateTo(() => {
-      const websites = appWebsites() || []
-      const preferredLanguageOrder = [...window.navigator.languages, 'en', null]
-
-      for (const language of preferredLanguageOrder) {
-        const site = websites.find((site) => siteHasLanguage(site, language))
-        if (site) return site
-      }
-
-      return websites[0] || null
-    })
+  if (await load(currentSiteId)) return
+
+  if (await load(isNoSitePresent)) {
+    ensureUserIsLoggedIn()
+    return
+  }
+
+  const site = await load(getPreferredSite)
+  if (!site) return
+
+  const siteUrl = await load(() => urlFor(site))
+  const { pathname, search, hash } = window.location
+  const path = pathname === '/' ? '' : pathname
+
+  window.location.assign(`${siteUrl}${path}${search}${hash}`)
```

Comment:
Will `siteUrl` always contain a trailing `/`? In your prior code you checked for it to end with `/` or not.

Reply:
This time it's the other way around: pathname will always **start** with a `/` (except if it's empty)

Reply:
But with `siteUrl` ends with a `/` and `path` starts with a `/` you will have `//` in your URL, right?

Reply:
Yes, but how can you make `siteUrl` end with `/`?